src/metrics.ts

```diff
@@ -129,79 +180,62 @@ export class MetricsProcessor implements MetricsProcessorInterface {
     const metricsData: MetricsData[] = [];
 
     // clone map and clear data
-    const clonedData = new Map(this.data);
-    this.data.clear();
-
-    for (const event of clonedData.values()) {
-      if (event.target && !event.target.anonymous) {
-        let targetAttributes: KeyValue[] = [];
-        if (event.target.attributes) {
-          targetAttributes = Object.entries(event.target.attributes).map(
-            ([key, value]) => {
-              const stringValue =
-                value === null || value === undefined
-                  ? ''
-                  : this.valueToString(value);
-              return { key, value: stringValue };
-            },
-          );
-        }
-
-        let targetName = event.target.identifier;
-        if (event.target.name) {
-          targetName = event.target.name;
-        }
-
-        const td: TargetData = {
-          identifier: event.target.identifier,
-          name: targetName,
-          attributes: targetAttributes,
-        };
-        targetData.push(td);
-      }
+    const clonedEvaluationAnalytics = new Map(this.evaluationAnalytics);
+    const clonedTargetAnalytics = new Map(this.targetAnalytics);
+    this.evaluationAnalytics.clear();
+    this.targetAnalytics.clear();
+    this.evaluationAnalyticsExceeded = false;
+    this.targetAnalyticsExceeded = false;
 
+    clonedEvaluationAnalytics.forEach((event) => {
```

Comment:
TOTALLY not an issue, but instead of creating a variable and then using a `forEach` to push to that variable. You can just swap to `map` and assign in one.
```suggestion
    const metricsData: MetricsData[] = clonedEvaluationAnalytics.map((event) => {
```

Reply:
I like 🚀  thanks.

Reply:
Dang - the `map` function isn't available on `Map` types.  I'd need to convert `clonedEvaluationAnalytics` to an array.  I'll opt to leave it as is.  But in principle I agree with the functional style here.

Reply:
Ah man... annoying. You could spread into an array or use `Array.from`. Don't worry about it, it's fine as-is.
```suggestion
    const metricsData: MetricsData[] = [...clonedEvaluationAnalytics.values()].map((event) => {
```